src/plugins/ProgressDrawer.js

```diff
@@ -22,41 +16,75 @@ export default class ProgressDrawer extends Plugin {
 
     // merge default options with the ones set by user
     this.opts = Object.assign({}, defaultOptions, opts)
+
+    this.el = this.render(this.core.state)
   }
 
-  render (state) {
-    return hx`<div>
-      ${state.bla}
-    </div>`
+  update (state) {
+    const newEl = this.render(state)
+    yo.update(this.el, newEl)
   }
 
-  // all actions should be in core, I guess
-  events () {
-    this.core.emitter.on('progress', function () {
-      this.updateState({ progress: loop.state.clickedTimes + 1 })
-    })
+  render (state) {
+    const selectedFiles = state.selectedFiles
+    const selectedFileCount = Object.keys(selectedFiles).length
+    const isSomethingSelected = selectedFileCount > 0
+    const autoProceed = this.core.opts.autoProceed
```

Comment:
You can destructure here with `const { autoProceed } = this.core.opts` to make this a little less long.  Same with `selectedFiles`, if you'd like.